### ISSUE TITLE
`UnstructuredDirectoryLoader` returns 403 - forbidden

### ISSUE THREAD

**Issue by karimStekelenburg** [hwchase17/langchainjs#1053]
I've been trying to use the `UnstructuredDirectoryLoader` to load in a directory that contains nested Markdown files. I've followed the instructions about the Docker container, and it seems to work for most docs. However, for a certain document, it seems that the Docker container is returning a 403 (forbidden) response code. There is nothing interesting about that specific document. The content is pure Markdown, and the file permissions are the same as for the rest. I ran it 4 times, and every time it breaks on the same file.

Any ideas on what might cause this?

![CleanShot 2023-04-29 at 21 56 03@2x](https://user-images.githubusercontent.com/18397186/235321933-eb555737-5322-4a7d-a4ef-ff20f4eb7786.png)

**Comment by nfcampos** [comment]
Can you share which URL you're passing to UnstructuredDirectoryLoader ?

**Comment by karimStekelenburg** [comment]
Hi there! In the meantime, I have figured out that this probably had to do with me using the hosted API (api.unstructured.io). I expect the 403 is due to a rate limit. After trying it with a Docker container on my machine, that problem is no longer there.

I am running into some other problems, though. Because I'm trying to import quite a big directory (~1k files), I get all sorts of fun issues like `JSON.stringify()` running out of memory and JSON payloads being too large when posting to the Chroma API.

After digging through the code, I saw some comments about streaming not being implemented yet. It may be worth mentioning somewhere in the docs that large directories can cause these types of issues.

Anyway, since these problems are not related to the `UnstructuredDirectoryLoader`, I'll close it for now!
